components/autocomplete/index.js

```diff
@@ -345,6 +357,7 @@ export class Autocomplete extends Component {
 				value: optionData,
 				label: completer.getOptionLabel( optionData ),
 				keywords: completer.getOptionKeywords ? completer.getOptionKeywords( optionData ) : [],
+				isDisabled: completer.isOptionDisabled ? completer.isOptionDisabled( optionData ) : false,
 			} ) );
 
 			const filteredOptions = filterOptions( this.state.search, keyedOptions );
```

Comment:
It may be a pathological case, but since we show a limited number of options at a time, it is technically possible for all shown options to be disabled. Maybe we should update filterOptions to sort disabled options to the end.

Reply:
Really like this idea, but let's punt on it until disabled items are a more common occurrence. Right now it's really only the More block case that we have to support.